test-five/react-app-cdk/lib/react-app-cdk-stack.ts

```diff
@@ -0,0 +1,88 @@
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+// import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
+import * as cloudfront_origins from 'aws-cdk-lib/aws-cloudfront-origins';
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+export interface ReactAppCdkProps extends cdk.StackProps{
+  stage: string;
+  path: string;
+}
+
+export class ReactAppCdkStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: ReactAppCdkProps) {
+    super(scope, id, props);
+
+    const { stage, path } = props;
+
+    // The code that defines your stack goes here
+
+    // example resource
+    // const queue = new sqs.Queue(this, 'ReactAppCdkQueue', {
+    //   visibilityTimeout: cdk.Duration.seconds(300)
+    // });
+
+    const cloudfrontOAI = new cloudfront.OriginAccessIdentity(this, 'cloudfront-OAI', {
+      comment: `OAI for react-app`
+    }
+    );
+
+    // create s3 bucket to hold build files from react-app
+    const bucket = new s3.Bucket(this, `react-app-${stage}`, {
+      bucketName: `react-app-${stage}-bucket`,
+      websiteIndexDocument: 'index.html',
+      websiteErrorDocument: 'index.html',
+      publicReadAccess: false,
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      autoDeleteObjects: true,
+    });
+
+    // attach bucket policy to grant access to cloudfront distro
+    bucket.addToResourcePolicy(
+      new cdk.aws_iam.PolicyStatement({
+        actions: ['s3:GetObject'],
+        resources: [bucket.arnForObjects('*')],
+        principals: [new iam.CanonicalUserPrincipal(cloudfrontOAI.cloudFrontOriginAccessIdentityS3CanonicalUserId)]
+      })
+    )
+
+    // create cloudfront distribution
+    const distribution = new cloudfront.Distribution(this, `react-app-${stage}-distro`, {
+      defaultRootObject: "index.html",
+      defaultBehavior: {
+        origin: new cloudfront_origins.S3Origin(bucket, {originAccessIdentity: cloudfrontOAI}),
+        compress: true,
+        allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
+        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.ALLOW_ALL,
+      },
+      priceClass: cloudfront.PriceClass.PRICE_CLASS_100,
+      comment: 'React App CDK Cloudfront Distribution'
+    })
+
+    // deploy react app to s3 bucket
+    new cdk.aws_s3_deployment.BucketDeployment(this, 'react-app-${stage}-deployment', {
+      destinationBucket: bucket,
+      sources: [cdk.aws_s3_deployment.Source.asset(path)],
+      cacheControl: [
```

Comment:
can you explain why you have added caching here and how it will be expired as pard of CD process?

Reply:
I've assumed that there may be changes to the react app codebase daily so have set the cache control to 1 day. This cache control minimises the data transfer between cloudfront and s3 which will help reduce cost. Caching can also improve performance due to faster load times from cloudfront cache. 

If changes are deployed to the react app more frequently (multiple times a day), a cloudfront invalidation step can be added to the cd process to ensure that the cloudfront cache is purged and the latest changes from the s3 bucket are served to the users